WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -196,6 +196,70 @@ def register
     end
   end
 
+  def process_payment
+    competition = competition_from_params
+    if current_user
+      registrations = competition.registrations
+      registration = registrations.find_by_user_id!(current_user.id)
+    end
+    token = params[:stripeToken]
+    Stripe.api_key = ENVied.STRIPE_API_KEY
+
+    begin
```

Comment:
I think it's better to have the `rescue` clauses at the end of the function and not have the `begin/end` block. This removes some code clutter, separates error handling from the regular flow, and makes you not have to remember to put a `return` at the end of each rescue.

Same goes for the one in `refund_payment`